pkg/oauth/apiserver/registry/oauthtokenreview/rest/tokenreviewwrapper.go

```diff
@@ -0,0 +1,94 @@
+package rest
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/emicklei/go-restful"
+
+	authenticationv1 "k8s.io/api/authentication/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apiserver/pkg/authentication/authenticator"
+	"k8s.io/apiserver/pkg/endpoints/handlers/responsewriters"
+	"k8s.io/kubernetes/pkg/apis/authentication"
+	authenticationv1internal "k8s.io/kubernetes/pkg/apis/authentication/v1"
+	ktokenreview "k8s.io/kubernetes/pkg/registry/authentication/tokenreview"
+)
+
+var (
+	scheme  = runtime.NewScheme()
+	codecs  = serializer.NewCodecFactory(scheme)
+	encoder = codecs.LegacyCodec(
+		authentication.SchemeGroupVersion,
+		authenticationv1.SchemeGroupVersion,
+	)
+)
+
+func init() {
+	utilruntime.Must(authentication.AddToScheme(scheme))
+	utilruntime.Must(authenticationv1.AddToScheme(scheme))
+}
+
+type TokenReviewRESTWrapper struct {
+	wrapped *ktokenreview.REST
+}
+
+func NewTokenReviewRESTWrapper(authenticator authenticator.Request) *TokenReviewRESTWrapper {
+	return &TokenReviewRESTWrapper{
+		wrapped: ktokenreview.NewREST(authenticator, []string{}),
+	}
+}
+
+func (tw *TokenReviewRESTWrapper) ServeHTTP(r *restful.Request, w *restful.Response) {
```

Comment:
this looks differently to other rest implementations. Would expect something like in k/k/pkg/registry/authz/subjectaccessreview.